docs/source/60_building_apps.md

```diff
@@ -1,3 +1,250 @@
+
+ADAM ACTIONS
+           myCommand : My command example.
+          countKmers : Counts the k-mers/q-mers from a read dataset.
+    countContigKmers : Counts the k-mers/q-mers from a read dataset.
+...
+
+$ ./bin/adam-submit myCommand input.foo
+```
+
+Then consider making a pull request to include the new command in ADAM!
+
+
+## Extend the ADAM CLI by adding new commands in an external repository {#external-commands}
+
+To extend the ADAM CLI by adding new commands in an external repository,
+instead of editing `ADAMMain` to add new commands, create a new main class
+that delegates to `ADAMMain` and provides additional command(s) via its
+constructor.
+
+```scala
+import org.bdgenomics.adam.cli.{ ADAMMain, CommandGroup }
+import org.bdgenomics.adam.cli.ADAMMain.defaultCommandGroups
+
+object MyCommandsMain {
+  def main(args: Array[String]) {
+    val commandGroup = List(CommandGroup("MY COMMANDS", List(MyCommand1, MyCommand2)))
+    new ADAMMain(defaultCommandGroups.union(commandGroup))(args)
+  }
+}
+```
+
+Build the project and run the new external commands via `adam-submit`,
+specifying `ADAM_MAIN` environment variable as the new main class,
+and providing the jar file in the Apache Spark `--jars` argument.
+
+Note the `--` argument separator between Apache Spark arguments and
+ADAM arguments.
+
+```bash
+$ ADAM_MAIN=MyCommandsMain \
+  adam-submit \
+  --jars my-commands.jar \
+  -- \
+  --help
+
+Using ADAM_MAIN=MyCommandsMain
+Using SPARK_SUBMIT=/usr/local/bin/spark-submit
+
+       e         888~-_          e             e    e
+      d8b        888   \        d8b           d8b  d8b
+     /Y88b       888    |      /Y88b         d888bdY88b
+    /  Y88b      888    |     /  Y88b       / Y88Y Y888b
+   /____Y88b     888   /     /____Y88b     /   YY   Y888b
+  /      Y88b    888_-~     /      Y88b   /          Y888b
+
+Usage: adam-submit [<spark-args> --] <adam-args>
+
+Choose one of the following commands:
+...
+
+MY COMMANDS
+          myCommand1 : My command example 1.
+          myCommand2 : My command example 2.
+
+$ ADAM_MAIN=MyCommandsMain \
+  adam-submit \
+  --jars my-commands.jar \
+  -- \
+  myCommand1 input.foo
+```
+
+A complete example of this pattern can be found in the
+[heuermh/adam-commands](https://github.com/heuermh/adam-examples) repository.
+
+
+## Use ADAM as a library in new applications {#library}
+
+To use ADAM as a library in new applications:
+
+Create a main object and handle command line arguments. Feel free to use the
+[args4j library](http://www.slf4j.org/) or any other argument parsing library.
+
+```scala
+object MyExample {
+  def main(args: Array[String]) {
+    if (args.length < 1) {
+      System.err.println("at least one argument required, e.g. input.foo")
+      System.exit(1)
+    }
+```
+
+Create an Apache Spark configuration `SparkConf` and use it to create a new `SparkContext`.
+The following serialization configuration needs to be present to register ADAM classes.
```

Comment:
I'll open a PR against this. There's another way to do the serializer registration, e.g., if you need to add your own serializer registrations. That being said, your approach is likely the common case.